### DISCUSSION TITLE
Hunt interface practically stopped working after update to 2.3.130

### DISCUSSION THREAD

**Discussion by Jaap79** [Security-Onion-Solutions/securityonion#8099] [Unsupported Versions] [answered]
Hi,

I just updated our grid to SO 2.3.130.. everything seemed to work fine but I'm having some issues pivoting from Dashboards or the Alerts interface to Hunt. 

The error "The search query encountered a failure within the Elasticsearch cluster. Check SOC logs for details." keeps popping up. For instance, when I use the Dashboard and click "Firewall" -> Actions -> Hunt, this error shows up. However, if I enter **_event.datasource:"firewall" | groupby event.module event.dataset_**   it all seems to work fine. 

When I pivot to the same dataset from, say, Alerts or the Hunt interface itself it will fail again with the "the search query encountered a failure [...]" error, so manually pivoting is not working for me.

If i tail -f the sensoroni-server.log in /opt/so/log/soc/ I see the following at the time of the query:

timestamp=2022-06-09T09:34:32.919737254Z level=warn message="Shard failure" reason="failed to create query: field expansion for [*] matches too many fields, limit: 3500, got: 4107" type=query_shard_exception

timestamp=2022-06-09T09:34:32.919807117Z level=warn message="Shard failure" reason="failed to create query: field expansion for [*] matches too many fields, limit: 3500, got: 4173" type=query_shard_exception

timestamp=2022-06-09T09:34:32.919828368Z level=warn message="Request did not complete successfully" error=ERROR_QUERY_FAILED_ELASTICSEARCH requestId=ce8e5b4d[...] etc.

Is there something I'm missing or should be changed?

**Comment by Jaap79** [comment]
Slight update:

It may have something to do with the max_clause_count .. .the default of 3500 apparently is no longer high enough. Increased it to 4500 and despite some issues after that (error 502..) it all seems to work now after a few restarts and highstate-calls.

Will keep this open for a few days to see if all issues are resolved.

**Comment by Jaap79** [comment]
Setting it to the required maximum (4300-some) will crash my Search Nodes even after adding another 8GB of RAM and 4 CPU's... (totalling 32GB / 16 vCPU for 2 nodes each). 

Kind of at a loss here. If I tune it back to the default 3500 it seems to run fine but the Hunt interface won't work. Setting it too high and I'll be rebooting 5x a day...

**Comment by dougburks** [comment, marked as answer]
For others that may come across this discussion:
https://docs.securityonion.net/en/2.3/elasticsearch.html#field-expansion-matches-too-many-fields

**Comment by Jaap79** [comment]
Slight update:

It seems like Playbook generates indices with a huge amount of fields. If I pick a random so-playbook-alerts-<date> I'm easily over 2600 fields (I copy the contents of the Mapping tab in Kibana and grep 'type' and wc -l the lines..). There's a whole bunch of those indices but could this be related in some way/could there be an index that has >4000 fields like this?
